### PR TITLE
fix(completion): Give completion for local variables when pointing to whitespace

### PR DESCRIPTION
**Pull request by Marwes**
Before the closest identifier to the left of the location would be completed which meant
```f#
test   abc
//   ^ completion here would complete `test` instead of listing local variables
```
This fix ensures that local variables are enumerated instead.

### CODE REVIEW

**Comment by Marwes** [comment]
While testing this I found that operators will be enumerated without parentheses. That is perfectly legal but it might not be what is expected.

```f#
test 
//    ^ completion here will enumerate '+', '-' etc 
// Is it better if it is '(+)', (-), etc ?
```

File: base/src/ast.rs [hunk 1]
```diff
@@ -261,7 +261,12 @@ impl<Id> LExpr<Id>
     pub fn span<'a>(&self, env: &(DisplayEnv<Ident = Id> + 'a)) -> Span {
         use self::Expr::*;
         let end = match self.value {
-            Identifier(ref id) => self.location.line_offset(CharPos(env.string(id).len())),
+            Identifier(ref id) => {
+                let name = env.string(id);
+                // Remove the `:<location>` suffix generated when renaming creates a unique name
+                let offset = name.split(':').next().unwrap_or(name).len();
+                self.location.line_offset(CharPos::from(offset))
+            }
```
Comment (Marwes) [inline base/src/ast.rs]: This is a workaround due to the current renaming behaviour. With #116 this is obviously not nececssary

**Comment by Marwes** [comment]
Copied the byte positions in the compleition tests from #115 to get the tests to pass again so I guess one of these is going to get a merge conflict.

**Comment by brendanzab** [comment]
You'll need to rebase this now - the completion API now uses byte positions, not locations.

**Comment by Marwes** [comment]
Only after #115 is merged, we can wait with merging this until after that as I only need to delete b576cf4d7f78241bbe751aa3bac2577dd170e6f1

**Comment by brendanzab** [comment]
Oh, durp, for some reason I thought I'd finished it!

**Pull request merged**